undefined reference to `dmlc::CustomLogMessage::Log

I built xgboost with cmake:
```
mkdir build
cd build
cmake .. -DUSE_CUDA=ON -DGOOGLE_TEST=ON -DUSE_AVX=OFF -DUSE_NCCL=OFF
make
```
Platform : Fedora 27 x86-64
GCC: gcc (GCC) 7.3.1 20180712 (Red Hat 7.3.1-6)
ld: GNU ld version 2.29-13.fc27
CUDA: 9.2

Here is the related error output:

> [ 85%] Built target testxgboost
[ 86%] Linking CXX executable dmlc_unit_tests
CMakeFiles/dmlc_unit_tests.dir/unittest_any.cc.o: In function `Any_json_Test::TestBody()':
unittest_any.cc:(.text+0x36ac): undefined reference to `dmlc::CustomLogMessage::Log(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/dmlc_unit_tests.dir/unittest_any.cc.o: In function `dmlc::CustomLogMessage::~CustomLogMessage()':
unittest_any.cc:(.text._ZN4dmlc16CustomLogMessageD2Ev[_ZN4dmlc16CustomLogMessageD5Ev]+0xd4): undefined reference to `dmlc::CustomLogMessage::Log(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/dmlc_unit_tests.dir/unittest_lockfree.cc.o: In function `Lockfree_ConcurrentQueue_Test::TestBody()':
unittest_lockfree.cc:(.text+0x13a4): undefined reference to `dmlc::CustomLogMessage::Log(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
unittest_lockfree.cc:(.text+0x26b6): undefined reference to `dmlc::CustomLogMessage::Log(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/dmlc_unit_tests.dir/unittest_lockfree.cc.o: In function `Lockfree_BlockingConcurrentQueue_Test::TestBody()':
unittest_lockfree.cc:(.text+0x4993): undefined reference to `dmlc::CustomLogMessage::Log(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/dmlc_unit_tests.dir/unittest_lockfree.cc.o:unittest_lockfree.cc:(.text+0x5c5d): more undefined references to `dmlc::CustomLogMessage::Log(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)' follow
collect2: error: ld returned 1 exit status
make[2]: *** [dmlc-core/test/unittest/CMakeFiles/dmlc_unit_tests.dir/build.make:311: dmlc-core/test/unittest/dmlc_unit_tests] Error 1
make[1]: *** [CMakeFiles/Makefile2:1332: dmlc-core/test/unittest/CMakeFiles/dmlc_unit_tests.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,10 @@ if(USE_AVX)
   add_definitions(-DXGBOOST_USE_AVX)
 endif()
 
+# dmlc-core
+add_subdirectory(dmlc-core)
+set(LINK_LIBRARIES dmlc rabit)
+
 # enable custom logging
 add_definitions(-DDMLC_LOG_CUSTOMIZE=1)
 
@@ -107,12 +111,6 @@ if(MINGW OR R_LIB)
 else()
   add_library(rabit STATIC ${RABIT_SOURCES})
 endif()
-
-
-# dmlc-core
-add_subdirectory(dmlc-core)
-set(LINK_LIBRARIES dmlc rabit)
-
 
 if(USE_CUDA)
   find_package(CUDA 8.0 REQUIRED)

